extensions/functions/src/main/java/io/crate/window/OffsetValueFunctions.java

```diff
@@ -149,8 +157,24 @@ private Object getValueAtTargetIndex(@Nullable Integer targetIndex,
             if (rowCells == null) {
                 throw new IndexOutOfBoundsException();
             }
-            row = new RowN(rowCells);
-            indexToRow.putIfAbsent(targetIndex, row);
+            row = indexToRow.get(targetIndex);
+            if (row == null) {
+                row = new RowN(rowCells);
+                // rowCells is already accounted, RowN is a wrapper to make expressions work.
+                // We account only for RowN shallow size and entry size.
+                long shallowRowSize = RamUsageEstimator.shallowSizeOfInstance(RowN.class);
+
+                // Entry calculation is based on IntObjectHashMap.ramBytesUsed()
+                int sizeBeforePut = indexToRow.size();
+                indexToRow.put(targetIndex, row);
+                int deltaSize = indexToRow.size() - sizeBeforePut;
+                long mapExpansionSize =
```

Comment:
As @matriv pointed out, expansion calculation probably shouldn't be based on size() but on capacity as size diff is always 1. Will double check, thanks for catching!

Reply:
as suggested, using `Accountable` method to compute, not checking neither size nor capacity


pushed https://github.com/crate/crate/pull/16054/commits/bafa7e86af793fa64728c580fe7712cf2938d641